src/main/java/core/basesyntax/dao/FruitStorageDaoImpl.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import java.util.Set;
+
+public class FruitStorageDaoImpl implements FruitStorageDao {
+    @Override
+    public boolean setBalance(Fruit fruit, int quantity) {
+        Storage.fruits.put(fruit, quantity);
+        return Storage.fruits.get(fruit) == quantity;
+    }
+
+    @Override
+    public boolean add(Fruit fruit, int quantity) {
+        int oldValue = Storage.fruits.get(fruit);
```

Comment:
There is a potential NullPointerException if the fruit does not exist in the Storage yet. You should check if `Storage.fruits.get(fruit)` returns null before trying to use its value.